File: tslang/src/hashmap/left-join/left-join.ts
const leftJoin = (
	hashMapOne: Map<string, string>,
	hashMapSecond: Map<string, string>,
): (string | null)[][] => {
	const result: (string | null)[][] = [];

	for (const [key, value] of hashMapOne) {
		const secondItem = hashMapSecond.get(key) || null;
		result.push([key, value, secondItem]);
	}

	return result;
};

export { leftJoin };

File: golanglang/internal/hashmap/left-join/left-join.go
package leftjoin

func DoLeftJoin(hashMapOne map[string]string, hashMapSecond map[string]string) [][]string {
	asSlice := make([][]string, len(hashMapOne))

	index := 0

	for key, value := range hashMapOne {

		asSlice[index] = append(asSlice[index], key, value, hashMapSecond[key])
		index++
	}

	return asSlice
}
